astropy/coordinates/representation/cylindrical.py

```diff
@@ -154,7 +154,7 @@ class CylindricalDifferential(BaseDifferential):
 
     base_representation = CylindricalRepresentation
 
-    def __init__(self, d_rho, d_phi=None, d_z=None, copy=False):
+    def __init__(self, d_rho, d_phi=None, d_z=None, copy=COPY_IF_NEEDED):
```

Comment:
This is actually really weird - why would the `Differential` have `copy=False` as a default when the representation does not (and indeed all representations and differentials have `copy=True`, though mostly implicitly via `BaseRep*`; but a concrete example is `PhysicsSphericalDifferential`).

It maybe should be in a different PR, but let's at least not mention this in the changelog entry...

Reply:
How about I revert this bit here and open a dedicated issue for it ?

Reply:
Yes, good idea, let's just have a PR that changes it to `True` - this is the default for essentially all astropy classes (except `Masked`, where `False` makes much more sense).

Reply:
done in #16198

Reply:
(and reverted here)

Reply:
No, it shouldn't be reverted - just use `copy=True` here!

Reply:
Oops, sorry, commenting on something that's out of date! All OK.